akka-remote/src/main/scala/akka/remote/serialization/DaemonMsgCreateSerializer.scala

```diff
@@ -61,7 +64,21 @@ private[akka] class DaemonMsgCreateSerializer(val system: ExtendedActorSystem) e
           .setClazz(props.clazz.getName)
           .setDeploy(deployProto(props.deploy))
         props.args map serialize foreach builder.addArgs
-        props.args map (a ⇒ if (a == null) "null" else a.getClass.getName) foreach builder.addClasses
+        props.args.map { a ⇒
+          val argClassName =
+            if (a == null) "null"
+            else {
+              val className = a.getClass.getName
+              if (scala212OrLater && a.getClass.isInstanceOf[Serializable] && className.contains("$Lambda$")) {
+                // The serialization of the parameters is based on passing class name instead of
```

Comment:
the problem is not serialization of the lambdas, it's our way of using the class of the lamda
```
[JVM-2] java.lang.ClassNotFoundException: akka.remote.NewRemoteActorSpec$$$Lambda$459/1116094714
[JVM-2] 	at java.lang.Class.forName0(Native Method)
[JVM-2] 	at java.lang.Class.forName(Class.java:348)
[JVM-2] 	at akka.actor.ReflectiveDynamicAccess.$anonfun$getClassFor$1(ReflectiveDynamicAccess.scala:21)
[JVM-2] 	at akka.actor.ReflectiveDynamicAccess$$Lambda$135/251210093.apply(Unknown Source)
[JVM-2] 	at scala.util.Try$.apply(Try.scala:209)
[JVM-2] 	at akka.actor.ReflectiveDynamicAccess.getClassFor(ReflectiveDynamicAccess.scala:20)
[JVM-2] 	at akka.remote.serialization.DaemonMsgCreateSerializer.deserialize(DaemonMsgCreateSerializer.scala:118)
```

This is before deserialization, when we try `getClassFor` on the lamda class name